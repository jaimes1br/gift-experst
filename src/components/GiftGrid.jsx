import PropTypes from 'prop-types';
import { useFetchGifts } from "../hooks/useFetchGifts";
import { GridItem } from "./GridItem";

export const GiftGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifts( category );
    
    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className="card-grid">
                {
                    images.map((img) => (
                        <GridItem 
                            key={ img.id } 
                            { ...img }
                        />
                    ))
                }
            </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired,
}