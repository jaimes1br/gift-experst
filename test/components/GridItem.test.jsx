import { render, screen } from "@testing-library/react"
import { GridItem } from "../../src/components/GridItem"

describe('Test on <GridItem/>', () => {
  
    const title = 'One piece';
    const url = 'https://one-piece.com/luffy.jpg';

    test('should match with snapshot', () => {
        
        const { container } = render(<GridItem title={ title } url={ url }/>);
        expect ( container ).toMatchSnapshot();
    });
   
    test('should show the image and alt text', () => {
        
        render(<GridItem title={ title } url={ url }/>);

        // expect(screen.getByRole('img').src).toBe( url );
        // expect(screen.getByRole('img').alt).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('should show the title in the component', () => {
        
        render(<GridItem title={ title } url={ url }/>);
        expect( screen.getByText( title )).toBeTruthy();
    });        
});
