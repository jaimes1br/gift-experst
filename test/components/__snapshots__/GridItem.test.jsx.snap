// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GridItem/> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One piece"
      src="https://one-piece.com/luffy.jpg"
    />
    <p>
      One piece
    </p>
  </div>
</div>
`;
